### PR TITLE
Added simple assumption tests to check for NC pins

### PR DESCRIPTION
**Pull request by mbartling**
.Added simple assumption tests to check for NC pins. Should help catch some issues early on when adding new targets

Github Issue (Enhancement): https://github.com/ARMmbed/ci-test-shield/issues/18

### CODE REVIEW

**Comment by BlackstoneEngineering** [comment]
@0xc0170 If you approve LGTM ?

**Comment by 0xc0170** [comment]
@mbartling Please split code style changes to the code changes, it creates too much noise.

File: mbed_app.json [hunk 1]
```diff
@@ -27,6 +27,12 @@
         "AIN_4": "A4",
         "AIN_5": "A5",
         "AOUT":  "A5",
+        "AOUT_0": "A0",
```
Comment (0xc0170) [inline mbed_app.json]: Are really all Ax pins capable of DAC?

AOUT should be removed to be consistent ? Pins Ax start from 0 for instance

Comment (0xc0170) [inline mbed_app.json]: Are all this AOUT_x pins really analog out? I could not find this in any specifications. For instance, for  some targets like Freescale, they have just one DAC pin.

Comment (mbartling) [inline mbed_app.json]: This was a mistake on my part. I misread the documentation.

**Comment by 0xc0170** [comment]
@mbartling Any update? This would be nice to have in!

**Comment by 0xc0170** [comment]
@mbartling bump?

File: TESTS/assumptions/AnalogIn/AnalogIn.cpp [hunk 2]
```diff
@@ -55,24 +55,37 @@ void Can_AnalogIn_and_DigitalOut()
...
-    return STATUS_CONTINUE;
+    return STATUS_CONTINUE;	
 }
 
+
 // Test cases
 // TODO: take pin names from config file or generate from pinmap file
 Case cases[] = {
     Case("AnalogIn - existence of `DEVICE_ANALOGIN` macro", Macro_Test,greentea_failure_handler),
+	Case("AnalogIn - is pin 0 connected?", test_AnalogIn_NC<MBED_CONF_APP_AIN_0>, greentea_failure_handler),
```
Comment (0xc0170) [inline TESTS/assumptions/AnalogIn/AnalogIn.cpp]: tabs, as cases are misaligned?

File: TESTS/assumptions/AnalogOut/AnalogOut.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#include "unity.h"
+#include "utest.h"
+
+
+using namespace utest::v1;
+
+template <PinName d_pin>
+void test_AnalogOut_NC()
+{
+	TEST_ASSERT_MESSAGE(d_pin != NC, "Pin is NC");
```
Comment (0xc0170) [inline TESTS/assumptions/AnalogOut/AnalogOut.cpp]: 4 spaces to be used please

**Comment by mbartling** [comment]
Fixed the indentation issues

**Pull request merged**